ironfish-cli/src/commands/config/edit.ts

```diff
@@ -48,17 +79,34 @@ export class EditCommand extends IronfishCommand {
     const filePath = path.join(folderPath, DEFAULT_CONFIG_NAME)
 
     await writeFileAsync(filePath, output)
+    const existingContent = await readFileAsync(filePath, { encoding: 'utf8' })
     const code = await launchEditor(filePath, this.sdk.config)
 
     if (code !== 0) {
       this.exit(code || undefined)
     }
 
     const content = await readFileAsync(filePath, { encoding: 'utf8' })
-    const config = JSONUtils.parse<Record<string, unknown>>(content)
 
-    await client.uploadConfig({ config })
-    this.log('Uploaded config successfully.')
-    this.exit(0)
+    try {
+      const config = JSONUtils.parse<Record<string, unknown>>(content)
+      this.log(jsonColorizer(content))
+      await ConfigOptionsSchema.validate(config, { strict: true })
+      await client.uploadConfig({ config })
+      this.log('Uploaded config successfully.')
+      this.exit(0)
+    } catch (err) {
+      if (err instanceof ParseJsonError) {
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        this.log(`${err} Not a Valid JSON, Aborting changes !!`)
+        await writeFileAsync(filePath, existingContent)
+        this.exit(1)
+      } else {
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        this.log(`${err}, Aborting changes !!`)
+        await writeFileAsync(filePath, existingContent)
+        this.exit(1)
+      }
+    }
```

Comment:
I don't think you read any of the code at all based on these changes.

1. There's already validation in the uploadConfig()
2. Why is this writing with existingContent to file path? It's a temporary file allocated in the os.tmpdir(). why would you need to revert it?
3. You don't understand how template strings work if you're needing to add `eslint-disable-next-line`. You're just doing this to ignore the error its telling you without fixing the actual error. You can't and should not pass objects into template expressions.
4. All of this catch logic is duplicating the logic in the base command
5. There's no need to catch the errors at all here as you're du;plicating

Reply:
1. I have seen the check in upload config, but does that stops you from saving incorrect value in config ? No

If you are open to discuss and suggest improvements - I can fix the code and maybe we can merge this ?